workflow-examples/src/main/java/com/redhat/parodos/examples/simple/LoggingWorkFlowTask.java

```diff
@@ -43,16 +45,30 @@ public class LoggingWorkFlowTask extends BaseInfrastructureWorkFlowTask {
 	@Override
 	public WorkReport execute(WorkContext workContext) {
 		log.info("Writing a message to the logs from: {}", getName());
-		if (getWorkFlowChecker() != null) {
-			workContext.put(WorkFlowConstants.WORKFLOW_CHECKER_ID, getWorkFlowChecker().getName());
+		try {
+			String userId = getParameterValue(workContext, "user-id");
+			String apiServer = getParameterValue(workContext, "api-server");
+			log.info("task parameter 'api-server' value in {} is {}", getName(), apiServer);
+			log.info("workflow parameter 'user-id' value in {} is {}", getName(), userId);
+			if (getWorkFlowChecker() != null) {
+				workContext.put(WorkFlowConstants.WORKFLOW_CHECKER_ID, getWorkFlowChecker().getName());
+			}
+			return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+		}
+		catch (Exception e) {
+			log.error("There was an issue with the task {}: {}", getName(), e.getMessage());
 		}
-		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+		return new DefaultWorkReport(WorkStatus.FAILED, workContext);
 	}
 
 	@Override
 	public List<WorkFlowTaskParameter> getWorkFlowTaskParameters() {
-		return List.of(WorkFlowTaskParameter.builder().key("api-server").description("The api server")
-				.type(WorkFlowTaskParameterType.URL).optional(false).build());
+		return List.of(
+				WorkFlowTaskParameter.builder().key("api-server").description("The api server")
+						.type(WorkFlowTaskParameterType.URL).optional(false).build(),
```

Comment:
How do we differentiate between optional and missing parameter and required and missing? Do we rely on user to verify whether it was provided?

Reply:
we used to have a feature allowing developer to define defaultValue for optional parameters.  I will try to restore it here